pallets/phala/src/stakepool.rs

```diff
@@ -132,26 +143,41 @@ pub mod pallet {
 				.as_secs()
 				.saturated_into::<u64>();
 
-			// TODO:
-			// 1) should we just stop some of workers rather than all of it
-			// 2) just iterate pools that contains waitting withdraw rather than all of it
-			for pid in 0..PoolCount::<T>::get() {
-				let pool_info = Self::ensure_pool(pid).expect("Stake pool doesn't exist; qed.");
-				if !pool_info.withdraw_queue.is_empty() {
-					// the front withdraw always the oldest one
-					if let Some(info) = pool_info.withdraw_queue.front() {
-						if (now - info.start_time)
-							> T::InsurancePeriod::get().saturated_into::<u64>()
-						{
-							// stop all worker in this pool
-							// TODO: only stop running workers?
-							for worker in pool_info.workers {
-								let miner: T::AccountId = pool_sub_account(pid, &worker);
-								let _ = <mining::pallet::Pallet<T>>::stop_mining(miner);
+			let mut t = WithdrawTimestamps::<T>::get();
+			// has waiting withdraw request
+			if !t.is_empty() {
+				// we just handle timeout request every block
+				while !t.is_empty()
+					&& now - t.front().unwrap() > T::InsurancePeriod::get().saturated_into::<u64>()
+				{
+					let pools = WithdrawPools::<T>::get(&t.front().unwrap())
+						.expect("Pool list doesn't exist; qed.");
+					for &pid in pools.iter() {
+						let pool_info =
+							Self::ensure_pool(pid).expect("Stake pool doesn't exist; qed.");
+						// if we check the pool withdraw_queue here, we don't have to remove a pool from WithdrawPools when
+						// a pool has handled their waiting withdraw request before timeout. Compare the IO performance we
+						// think remove pool from WithdrawPools would have more resource cost.
+						if !pool_info.withdraw_queue.is_empty() {
```

Comment:
When a pool is already force stopped due to "withdraw1", but in the queue there are "withdraw2", "withdraw3", etc. I think we should skip this pool in the future. Otherwise we waste time to stop them again and again

Reply:
Yes, i am considering giving pool a state that can identify it was stopped, then we can just skip here.